src/Compiler/NodeVisitor/Behavior/ExprTypeNodeVisitorBehavior.php

```diff
@@ -11,6 +11,7 @@ trait ExprTypeNodeVisitorBehavior
 {
     public function getType(Node $node): ?Type
     {
-        return $node->getAttribute(ExprTypeNodeVisitorInterface::TYPE_ATTRIBUTE);
+        $type = $node->getAttribute(ExprTypeNodeVisitorInterface::TYPE_ATTRIBUTE);
+        return $type instanceof Type ? $type : null;
```

Comment:
Would not MixedType be better here? Of do we need to know when type is for some reason not given?

Reply:
As I think about it explicit null is probable better